### ISSUE TITLE
Consider migrating CPU detection features to Google cpu_features (ARM runtime dispatch)

### ISSUE THREAD

**Issue by lemire** [simdjson/simdjson#1047]
Though it is not yet feature complete, Google has a neat library to identify CPU features...

https://github.com/google/cpu_features/

**Comment by jkeiser** [comment]
This supports complementary compile time detection as well, which would enable some scenarios like excluding kernels that can't possibly be selected (if you compile for haswell, your program won't run on westmere) and "local architecture only" mode.

**Comment by lemire** [comment]
Right. The downside is that it might make our build far more complicated.

**Comment by aqrit** [comment]
Unfortunately, [broken](https://github.com/google/cpu_features/issues/4) since day 1, 2+ years ago, seems like a red flag.

**Comment by jkeiser** [comment]
That issue is likely something we could fix upstream ourselves with low effort ... but we need to see whether it's a signal that the library isn't well-maintained/supported.

**Comment by gchatelet** [comment]
I believe [the issue](https://github.com/google/cpu_features/issues/4) is now fixed on master.
There is still a slight chance a feature is not detected as there doesn't seem to be a way to detect `ssse3`, `sse4.1`, `sse4.2`  on Windows platforms for pre-AVX cpus.

Feel free to create bugs if needed.

**Comment by lemire** [comment]
@gchatelet Great.

**Comment by lemire** [comment]
Closing as not planned. simdjson is intentionally dependency-free, and adding cpu_features would complicate the build and the single-header distribution. We have kept extending our own detection code instead; for example, arm64 SVE runtime detection landed in #2867.
